AlphaWallet/Gas/Models/GasNowGasPriceEstimator.swift

```diff
@@ -11,3 +12,54 @@ class GasNowGasPriceEstimator {
         }
     }
 }
+
+class EtherscanGasPriceEstimator {
+
+    struct EtherscanPriceEstimatesResponse: Decodable {
+        let result: EtherscanPriceEstimates
+    }
+
+    static func supports(server: RPCServer) -> Bool {
+        return server.etherscanGasPriceEstimatesURL != nil
+    }
+
+    func fetch(server: RPCServer) -> Promise<GasNowPriceEstimates> {
+        struct AnyError: Error {}
+        guard let url = server.etherscanGasPriceEstimatesURL else {
+            return .init(error: AnyError())
+        }
+
+        return Alamofire.request(url, method: .get).responseDecodable(EtherscanPriceEstimatesResponse.self).compactMap { response in
+            EtherscanPriceEstimates.bridgeToGasNowPriceEstimates(for: response.result)
+        } 
+    }
+}
+
+fileprivate extension RPCServer {
+    var etherscanGasPriceEstimatesURL: URL? {
+        switch self {
+        case .main:
+            return URL(string: "https://api-cn.etherscan.com/api?module=gastracker&action=gasoracle&apikey=1PX7RG8H4HTDY8X55YRMCAKPZK476M23ZR")
```

Comment:
Can we form the URL with `etherscanApiRoot` like in `RPCServer.etherscanURLForGeneralTransactionHistory`?